### DISCUSSION TITLE
Animation Implicit Rules

### DISCUSSION THREAD

**Discussion by zyyv** [unocss/unocss#599] [Q&A]
In daily development, we use animation to follow implicit rules

```
animation: name duration timing-function delay ...
```

so, should `/^animate-(.+)$/` also follow the rules above ?

like: 
```
animate-[pulse_1s_ease-in-out] // ✅

animate-[1s_pluse_ease-in-out] // Although it can be parsed, it does not actually conform to the implicit rules
```

### Why am I asking this question
Now, in unocss@v0.24.3. i found  `/^animate-(.+)$/` should be optimized.

`animate-fade-in` is different from `animate-[fade-in]`, `animate-[fade-in]` missing `keyframes`

if unocss  follow implicit rules, we can get the keyframe name in ` h.bracket.cssvar(name)!.split(' ')[0]`

it seems like easyily

@chu121su12

**Comment by chu121su12** [comment]
I think the bracket & css-var rule (`$xxx`) should be used as a last resort to add unaccounted values as class. I'm sure parsing for correct animation syntax is good dx, but lightly skimming teh specs, the syntax is a bit complex https://developer.mozilla.org/en-US/docs/Web/CSS/animation#formal_syntax

As for missing components, there's complementary rules to add them:

`animate-keyframes-x`/`keyframes-x`:
https://github.com/unocss/unocss/blob/84b1fc3a8b01dc61d239628f8b0777e0a6706a55/packages/preset-wind/src/rules/animation.ts#L6-L10

Others:
https://github.com/unocss/unocss/blob/84b1fc3a8b01dc61d239628f8b0777e0a6706a55/packages/preset-wind/src/rules/animation.ts#L23-L38


In your example, the syntax may be:
`animate-[pulse_1s_ease-in-out] keyframes-pulse`

**Reply by zyyv** [reply]
Thanks for your reply.

My original intention was to want `animate-[pulse_1s_ease-in-out]` to include `keyframes-pulse`, like `animate-pulse`.
This way we can avoid writing `keyframes-pulse`

`animate-[pulse_1s_ease-in-out]` instead of `animate-[pulse_1s_ease-in-out] keyframes-pulse`
